tests/test_date_parser.py

```diff
@@ -198,3 +198,139 @@ def test_invalid_offset_timezone():
         ParseDate("2020-01-01 12:34:56 -5")  # Not enough digits
     assert "Invalid timezone name" in str(excinfo.value)
+
+
+def test_dst_handling():
+    """Test parsing dates near DST transitions."""
+    # Test a valid time (not in DST transition)
+    date = ParseDate("2023-03-11 02:30:00 US/Pacific")
+    assert date.year == 2023
+    assert date.month == 3
+    assert date.day == 11
+    assert date.hour == 2
+    assert date.minute == 30
+
+    # Test a time after DST starts
+    date = ParseDate("2023-03-12 03:30:00 US/Pacific")
+    assert date.year == 2023
+    assert date.month == 3
+    assert date.day == 12
+    assert date.hour == 3
+    assert date.minute == 30
+
+    # Test the offset timezone handling instead of named timezone for non-DST date
+    date = ParseDate("2023-11-05 01:30:00 -0800")  # PST offset
+    assert date.year == 2023
+    assert date.month == 11
+    assert date.day == 5
+    assert date.hour == 1
+    assert date.minute == 30
+    # -0800 is 8 hours behind UTC
+    utc_offset = date.tzinfo.utcoffset(date)
+    assert utc_offset.total_seconds() == -8 * 60 * 60
+
+
+def test_format_iso_date():
+    """Test ISO date formatting converts to UTC."""
+    # Create a date in US/Pacific timezone
+    pacific_tz = timezone("US/Pacific")
+    local_date = pacific_tz.localize(datetime.datetime(2023, 1, 15, 14, 30, 0))
+
+    # Format as ISO date (should convert to UTC)
+    iso_date = FormatISODate(local_date)
+
+    # US/Pacific is UTC-8 in January, so 14:30 should become 22:30 UTC
+    assert "2023-01-15T22:30:00Z" in iso_date
+
+
+def test_format_html_date():
+    """Test HTML date formatting preserves timezone."""
+    # Create a date in Europe/Paris timezone
+    paris_tz = timezone("Europe/Paris")
+    local_date = paris_tz.localize(datetime.datetime(2023, 6, 1, 14, 30, 0))
+
+    # Format as HTML date
+    html_date = FormatHtmlDate(local_date)
+
+    # Check time and timezone are in the output
+    assert "14:30:00" in html_date
+    assert "CEST" in html_date or "CET" in html_date
+    # Check date formatting
+    assert "1<sup>st</sup> June 2023" in html_date
+
+
+def test_format_atom_dates_single_date():
+    """Test that FormatAtomDates properly formats a single date."""
+    date = utc.localize(datetime.datetime(2023, 1, 15, 14, 30, 0))
+    atom_date = FormatAtomDates([date])
+    expected = "<updated>2023-01-15T14:30:00Z</updated>"
+    assert atom_date == expected
+
+
+def test_format_atom_dates_multiple_dates():
+    """Test that FormatAtomDates properly formats multiple dates."""
+    date1 = utc.localize(datetime.datetime(2023, 1, 15, 14, 30, 0))
+    date2 = utc.localize(datetime.datetime(2023, 1, 16, 10, 0, 0))
+
+    atom_date = FormatAtomDates([date1, date2])
+    expected1 = "<published>2023-01-15T14:30:00Z</published>"
+    expected2 = "<updated>2023-01-16T10:00:00Z</updated>"
+
+    assert expected1 in atom_date
+    assert expected2 in atom_date
+
+
+def test_timezone_conversion():
```

Comment:
This is not testing any of our code. please remove this test - it's only testing `pytz`